src/Fixer/Operator/BinaryOperatorSpacesFixer.php

```diff
@@ -656,6 +733,10 @@ private function injectAlignmentPlaceholdersForArrow(Tokens $tokens, int $startA
             // no need to analyze for `isBinaryOperator` (always true), nor if part of declare statement (not valid PHP)
             // there is also no need to analyse the second arrow of a line
             if ($token->isGivenKind(T_DOUBLE_ARROW) && $newLineFoundSinceLastPlaceholder) {
+                if ($yieldFoundSinceLastPlaceholder) {
```

Comment:
To handle https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6716#issuecomment-1364486054

When a yield was previously use, it's another scope/level.